Tests/AsyncHTTPClientTests/HTTPClientTests.swift

```diff
@@ -901,6 +886,75 @@ class HTTPClientTests: XCTestCase {
         }
     }
 
+    func testStressGetHttps() throws {
+        let httpBin = HTTPBin(ssl: true)
+        let httpClient = HTTPClient(eventLoopGroupProvider: .createNew,
+                                    configuration: HTTPClient.Configuration(certificateVerification: .none))
+        defer {
+            XCTAssertNoThrow(try httpClient.syncShutdown())
+            XCTAssertNoThrow(try httpBin.shutdown())
+        }
+
+        let eventLoop = httpClient.eventLoopGroup.next()
+        let requestCount = 200
+        var futureResults = [EventLoopFuture<HTTPClient.Response>]()
+        for _ in 1...requestCount {
+            let req = try HTTPClient.Request(url: "https://localhost:\(httpBin.port)/get", method: .GET, headers: ["X-internal-delay": "100"])
+            futureResults.append(httpClient.execute(request: req))
+        }
+        XCTAssertNoThrow(try EventLoopFuture<HTTPClient.Response>.andAllSucceed(futureResults, on: eventLoop).wait())
+    }
+
+    func testResponseDelayGet() throws {
+        let httpBin = HTTPBin(ssl: false)
+        let httpClient = HTTPClient(eventLoopGroupProvider: .createNew,
+                                    configuration: HTTPClient.Configuration(certificateVerification: .none))
+        defer {
+            XCTAssertNoThrow(try httpClient.syncShutdown())
+            XCTAssertNoThrow(try httpBin.shutdown())
+        }
+
+        let req = try HTTPClient.Request(url: "http://localhost:\(httpBin.port)/get", method: .GET, headers: ["X-internal-delay": "2000"], body: nil)
+        let start = Date()
+        let response = try! httpClient.execute(request: req).wait()
+        XCTAssertEqual(Date().timeIntervalSince(start), 2, accuracy: 0.25)
+        XCTAssertEqual(response.status, .ok)
+    }
+
+    func testIdleTimeoutNoReuse() throws {
+        let httpBin = HTTPBin(ssl: false)
+        let httpClient = HTTPClient(eventLoopGroupProvider: .createNew,
+                                    configuration: HTTPClient.Configuration(certificateVerification: .none))
+        defer {
+            XCTAssertNoThrow(try httpClient.syncShutdown())
+            XCTAssertNoThrow(try httpBin.shutdown())
+        }
+        var req = try HTTPClient.Request(url: "http://localhost:\(httpBin.port)/get", method: .GET)
+        XCTAssertNoThrow(try httpClient.execute(request: req, deadline: .now() + .seconds(2)).wait())
+        req.headers.add(name: "X-internal-delay", value: "2500")
+        try httpClient.eventLoopGroup.next().scheduleTask(in: .milliseconds(250)) {}.futureResult.wait()
+        XCTAssertNoThrow(try httpClient.execute(request: req).timeout(after: .seconds(5)).wait())
+    }
+
+    func testStressGetClose() throws {
```

Comment:
This sometimes fails (currently at https://github.com/swift-server/async-http-client/pull/105/commits/5eee6110a248d3708187be158870a7f53302042c) must investigate this issue.

Reply:
The failures I saw are likely linked to variations in response time that made it exceed the 3 seconds timeout, another possible cause could be synchronization issues we already spoke about but they will be fixed.